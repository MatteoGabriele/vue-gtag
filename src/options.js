import { mergeDeep } from "@/utils";

export const getDefaultParams = () => ({
  bootstrap: true,
  onReady: null,
  onError: null,
  onBeforeTrack: null,
  onAfterTrack: null,
  pageTrackerTemplate: null,
  customResourceURL: "https://www.googletagmanager.com/gtag/js",
  customPreconnectOrigin: "https://www.googletagmanager.com",
  deferScriptLoad: false,
  pageTrackerExcludedRoutes: [],
  pageTrackerEnabled: true,
  enabled: true,
  disableScriptLoad: false,
  pageTrackerScreenviewEnabled: false,
  appName: null,
  pageTrackerUseFullPath: false,
  pageTrackerPrependBase: true,
  pageTrackerSkipSamePath: true,
  globalDataLayerName: "dataLayer",
  globalObjectName: "gtag",
  defaultGroupName: "default",
  includes: null,
  config: {
    id: null,
    params: {
      send_page_view: false,
    },
  },
});

let params = {};

export const setOptions = (options = {}) => {
  const defaultParams = getDefaultParams();
  params = mergeDeep(defaultParams, options);
};

export const getOptions = () => {
  return params;
};
