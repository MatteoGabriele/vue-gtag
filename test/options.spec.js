import { getOptions, setOptions, getDefaultParams } from "@/options";

describe("options", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test("getDefaultParams", () => {
    expect(getDefaultParams()).toMatchSnapshot();
  });

  test("set and get options", () => {
    setOptions({ config: { id: 1 } });
    expect(getOptions()).toMatchSnapshot();
  });

  test("should get default options", () => {
    setOptions();
    expect(getOptions()).toEqual(getDefaultParams());
  });
});
