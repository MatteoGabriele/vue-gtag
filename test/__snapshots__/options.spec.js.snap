// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options getDefaultParams 1`] = `
Object {
  "appName": null,
  "bootstrap": true,
  "config": Object {
    "id": null,
    "params": Object {
      "send_page_view": false,
    },
  },
  "customPreconnectOrigin": "https://www.googletagmanager.com",
  "customResourceURL": "https://www.googletagmanager.com/gtag/js",
  "defaultGroupName": "default",
  "deferScriptLoad": false,
  "disableScriptLoad": false,
  "enabled": true,
  "globalDataLayerName": "dataLayer",
  "globalObjectName": "gtag",
  "includes": null,
  "onAfterTrack": null,
  "onBeforeTrack": null,
  "onError": null,
  "onReady": null,
  "pageTrackerEnabled": true,
  "pageTrackerExcludedRoutes": Array [],
  "pageTrackerPrependBase": true,
  "pageTrackerScreenviewEnabled": false,
  "pageTrackerSkipSamePath": true,
  "pageTrackerTemplate": null,
  "pageTrackerUseFullPath": false,
}
`;

exports[`options set and get options 1`] = `
Object {
  "appName": null,
  "bootstrap": true,
  "config": Object {
    "id": 1,
    "params": Object {
      "send_page_view": false,
    },
  },
  "customPreconnectOrigin": "https://www.googletagmanager.com",
  "customResourceURL": "https://www.googletagmanager.com/gtag/js",
  "defaultGroupName": "default",
  "deferScriptLoad": false,
  "disableScriptLoad": false,
  "enabled": true,
  "globalDataLayerName": "dataLayer",
  "globalObjectName": "gtag",
  "includes": null,
  "onAfterTrack": null,
  "onBeforeTrack": null,
  "onError": null,
  "onReady": null,
  "pageTrackerEnabled": true,
  "pageTrackerExcludedRoutes": Array [],
  "pageTrackerPrependBase": true,
  "pageTrackerScreenviewEnabled": false,
  "pageTrackerSkipSamePath": true,
  "pageTrackerTemplate": null,
  "pageTrackerUseFullPath": false,
}
`;
